src/collector.rs

```diff
@@ -1,3 +1,8 @@
+// Or we'll get deprecation warnings from our own deprecated type (seriously Rust?)
+#![allow(deprecated)]
```

Comment:
Why is this here?

Reply:
This was not added by me, I thought, it was in the Git history somewhere, but I can not find the commit. I will remove it then...

Reply:
I guess, it was added during rebase onto next yesterday by some of the commits.